### ISSUE TITLE
USWDS - Feature request: Improve usability of search "clear" button and evaluate search input type

### ISSUE THREAD

**Issue by aduth** [uswds/uswds#5277]
### Describe the bug

The [Search component](https://designsystem.digital.gov/components/search/) is not fully keyboard operable in all browsers. Specifically, Chrome will display an "X" button allowing a user to clear the field contents, but this button cannot be reached with a keyboard.

![image](https://user-images.githubusercontent.com/1779930/236900482-c6911a19-74de-4f6b-a901-43c2254bf774.png)

While this could arguably be blamed as a fault of Chrome's treatment of `[type=search]` fields, it is worth considering USWDS's responsibility in the practical effects of using these field semantics, and whether the value provided by them outweighs their potential for confusion or to block one's ability to complete tasks.

Related resources:

- ["Maybe Ignore type=search"](https://adrianroselli.com/2019/07/ignore-typesearch.html)
   - Makes an argument that `[type=search]` offers little to no benefit over `[type=text]`
- ["Why the GOV.UK Design System team changed the input type for numbers"](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/)
   - Prior art for GOV.UK's design system making a pragmatic implementation choice as a compromise given practical realities of trying to use the "semantic" option of `[type=number]`

Related Section 508 (WCAG AA) resources:

- [Success Criterion 2.1.1 Keyboard Operable](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)
- [F54: Failure of Success Criterion 2.1.1 due to using only pointing-device-specific event handlers (including gesture) for a function](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F54)

As additional context, this was flagged as a violation in a partner agency's accessibility audit of our application.

### Steps to reproduce the bug

1. In Chrome, go to https://designsystem.digital.gov/components/search/
2. Enter some text in one of the search fields
3. Activate the text field clear button ("X") that shows, using only the keyboard

### Expected Behavior

All interactive elements are keyboard operable.

Possible solutions could include:

- Avoid `[type=search]`, use `[type=text]` instead
- Find a way to hide the clear button using styles so that the experience is consistent regardless of input modality

### Related code

https://github.com/uswds/uswds/blob/9b40a93c1ba348d77a4b0c632928ddfc6383b149/packages/usa-search/src/usa-search.twig#L12

### System setup

- USWDS Version: 3.4.1
- Browser: Chrome 113.0.5672.63
- OS: macOS 12.6.5

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by aduth** [comment]
Here's a related Chromium bug which, if fixed, may also address the issue of keyboard accessibility in Chrome:

https://bugs.chromium.org/p/chromium/issues/detail?id=1147314

**Comment by amyleadem** [comment]
Thanks for submitting this and including helpful context in the issue description, @aduth. This is an interesting topic. 

Tagging this as "Needs: Discussion" because I think we need to decide:
- Can the component be considered keyboard operable if you cannot access the visual button with the keyboard but the button functionality could be achieved with common keyboard shortcuts (In this example, `esc` or `Ctrl+A`/`Cmd+A` + `delete`)?
- Considering the [article](https://adrianroselli.com/2019/07/ignore-typesearch.html) shared in the description, what value does `type="search"` add to this component? Would it be better replace it with a different input type?
- If we decide to keep the clear button, should it to be tab-able?

@amycole501 Assigning this to you and tagging this as a spike because it isn't immediately clear if there is an industry standard here.

**Comment by amycole501** [comment]
I agree this is an interesting topic. While trying to be helpful, these browsers have introduced a "partially helpful" delete feature that has instead created an accessibility issue.

- I would like to test hiding the clear function as I think this is our most equitable and simplest solution. (Question: Would we then need to retest this feature as browsers update or would this be a permanent solution?)

**Comment by amycole501** [comment]
While this does fall into the category of an accessibility review, we have determined the impact is low due to the following circumstances:

- The search component is keyboard-operable using standard keyboard commands.
- The action of deleting content from the field is operable in most browsers (not all users are affected).
- Changing the input type is a breaking change and requires more in-depth discussion.
- We are shifting focus from an accessibility bug to a usability feature request.
      - We will open the feature request with the goal of evaluating best practices for the clear button and the role of search input types.
- [Failure of WCAG 2.1.1](https://www.w3.org/WAI/WCAG21/Techniques/failures/F54) says a failure is a result of the "only mechanism available to invoke a function of the content". Since there is a "functionally equal"* option of clearing the field, this issue does not qualify as a failure.

*WCAG emphasizes the concept of "functional equivalence" meaning, in this case, that the "clear search" function can be achieved through a different method on the same page that does meet the success criterion.

**Comment by amycole501** [comment]
Related to https://github.com/uswds/uswds/issues/5834

**Comment by ryparker** [comment]
The earlier accessibility review found that users can clear the field with standard keyboard commands, so this was reclassified as a usability request. We aren't planning to change the input type or native clear button based on those findings alone.

Let's continue the usability question in a discussion. A current example of a task someone cannot complete, including their browser and assistive technology, would help us evaluate a change. The related report in #5834 still needs its own investigation.

**Comment by ryparker** [comment]
I'm closing this as not planned and continuing the usability question in [discussion #6971](https://github.com/uswds/uswds/discussions/6971). The earlier accessibility review found that standard keyboard commands can clear the field; we aren't planning to change the input type or native clear button based on those findings alone.

The discussion carries over the proposal and compatibility requirements. A current example of a task someone cannot complete, including their browser and assistive technology, would help us evaluate a change. #5834 still needs its own investigation.

GitHub's issue conversion didn't complete, so I created a linked discussion. The original report and comments remain here. Closing this doesn't mean the usability concern is resolved.
